crypto/ube/ube_test.cc

```diff
@@ -53,92 +33,39 @@ TEST_F(ubeTest, BasicTests) {
   ASSERT_EQ(current_generation_number, generation_number);
 }
 
-TEST_F(ubeTest, MockedMethodTestsSnapsafe) {
-
-  allowMockedUbe();
+static void MockedDetectionMethodTest(
+  std::function<void(uint32_t)> set_method_generation_number) {
 
   uint64_t generation_number = 0;
   uint64_t cached_generation_number = 0;
+  uint32_t mocked_generation_number = 0;
```

Comment:
Double checking: is the mocked number is a uint32_t because snapsafe requires using 32 bits, while our forking logic deals with 64 bit ints? Any reason why they're different?

Reply:
intentional because I didn't want to truncate the number, which would happen if starting with uint64_t and casting to uint32_t.